aleph-client/src/staking.rs

```diff
@@ -112,6 +112,21 @@ pub fn wait_for_next_era<C: AnyConnection>(connection: &C) -> anyhow::Result<Era
     wait_for_era_completion(connection, get_current_era(connection) + 1)
 }
 
+pub fn wait_for_at_least_era<C: AnyConnectionExt>(
+    connection: &C,
+    era: EraIndex,
+) -> anyhow::Result<EraIndex> {
+    let sessions_per_era: u32 = connection.read_constant(PALLET, "SessionsPerEra");
+    let mut current_era = get_era(connection, None);
+    while current_era < era {
+        let current_session = get_current_session(connection);
+        let first_session_in_era = current_session + (era - current_era) * sessions_per_era;
```

Comment:
`first_session_in_era = era_first_session_in_era + (era - current_era) * sessions_per_era`.
The current_session is not always the first session of the era

Reply:
That's why its name contains `..at_least`. If we want to make it more precise we should make that loop check each session (due to to possible `force_new_era` calls).

Reply:
Removed.